packages/amesos2/src/basker/basker_def.hpp

```diff
@@ -247,17 +255,18 @@ namespace BaskerClassicNS{
     /*End creating space for L and U*/
 
     /*Creating working space*/
-    Int *tptr;
+    Int *color, *pattern, *stack; 
     Entry *X;
-    //tptr = (Int *)   BASKERCALLOC( (ncol)+(4*nrow), sizeof(Int));
-    tptr = new Int[(ncol)+(4*nrow)]();
+    color = new Int[ncol]();
+    pattern = new Int[nrow]();
+    stack = new Int[2*nrow]();
```

Comment:
Thanks @hkthorn! Changes look reasonable to me, @srajama1 let us know if there's any unexpected consequence for splitting the workspace tptr allocation into the separate explicit allocations for color, pattern and stack (instead of previous behavior of later offsetting those pointers into the larger contiguous memory allocation of tptr)